### PR TITLE
Fix type convert in TiKV sink

### LINKED ISSUE #223: [BUG]  streaming mode with TiKV sink fail with int type column
**Describe the bug**
- TiBigData： master
- streaming mode, tikv sink 

TiKV sink fails to convert Flink row to TiDB row when the table contains any one of the following types column
- int
- bit
- tinyint
- smallint
- mediumint
- year

**What did you do**
1. create table
```
CREATE TABLE `test`.`t_src`  (
  `id` int(11) NOT NULL,
  `name` varchar(255) NULL DEFAULT NULL,
  PRIMARY KEY (`id`) 
) ;

CREATE TABLE `test`.`t_dst`  (
  `id` int(11) NOT NULL,
  `name` varchar(255) NULL DEFAULT NULL,
  PRIMARY KEY (`id`) 
) ;
```

2. Use Kafka source get change log from TiCDC
3. create catalog in sql-client
```
CREATE CATALOG `tidb` WITH (   'type' = 'tidb',   'tidb.database.url' = 'jdbc:mysql://127.0.0.1:4000?useSSL=false',   'tidb.username' = 'root',   'tidb.password' = '',   'tidb.streaming.source' = 'kafka',   'tidb.streaming.codec' = 'json',   'tidb.streaming.kafka.bootstrap.servers' = 'localhost:9092',   'tidb.streaming.kafka.topic' = 'cdc',   'tidb.streaming.kafka.group.id' = 'test_cdc_group',   'tidb.streaming.ignore-parse-errors' = 'true' ,'tidb.sink.impl'='TIKV')
```
4. execute SQL in flink sql-client 
```
INSERT INTO `tidb`.`test`.`t_dst` SELECT id,name FROM `tidb`.`test`.`t_src`
```
6. Use JDBC to insert data into flink_sink
```
INSERT into test.flink_sink VALUES (1,1)
```
7. you will find the ClassCastException is thrown

**Screenshots**
image

### PR DESCRIPTION
**Pull request by shiyuhang0**
## What is the purpose of the change
close #223

## Brief change log

- fix the TiDBRowConverter.toTiRow:  the error type convert will throw ClassCastException 
- fix the decode：Object to long will throw ClassCastException

## Verifying this change

- an IT has been added

## Does this pull request potentially affect one of the following parts:

- Dependencies (does it add or upgrade a dependency): (no)
- The public API: ( no)
- The runtime per-record code paths (performance sensitive): (yes)

## Documentation

- Does this pull request introduce a new feature? (no)

## JMH Test
jmh for TiDBRowConverter.toTiRow
- before this pr：60 ns/op (bigint type)
image

- in this pr：64 ns/op (int/tinyint/smallint/mudiumnint/year type)
image

jmh for convert object to long
-  convertLong test the method before this pr：  ` (long)obj` 
- parseLong test the method in this pr `Long.parseLong(obj.toString())`

```
Benchmark                                            (data)    Mode      Cnt        Score   Error  Units
ParseLongBenchMark.convertLong                            1  sample  1232931       26.119 ± 0.899  ns/op
ParseLongBenchMark.convertLong:convertLong·p0.00          1  sample                   ≈ 0          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.50          1  sample                41.000          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.90          1  sample                42.000          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.95          1  sample                42.000          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.99          1  sample                42.000          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.999         1  sample                84.000          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.9999        1  sample              6344.544          ns/op
ParseLongBenchMark.convertLong:convertLong·p1.00          1  sample            256256.000          ns/op
ParseLongBenchMark.convertLong                      1000000  sample  1223389       27.422 ± 0.904  ns/op
ParseLongBenchMark.convertLong:convertLong·p0.00    1000000  sample                   ≈ 0          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.50    1000000  sample                41.000          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.90    1000000  sample                42.000          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.95    1000000  sample                42.000          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.99    1000000  sample                42.000          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.999   1000000  sample                84.000          ns/op
ParseLongBenchMark.convertLong:convertLong·p0.9999  1000000  sample              9284.608          ns/op
ParseLongBenchMark.convertLong:convertLong·p1.00    1000000  sample            191232.000          ns/op
ParseLongBenchMark.parseLong                              1  sample  1461371       33.283 ± 3.144  ns/op
ParseLongBenchMark.parseLong:parseLong·p0.00              1  sample                   ≈ 0          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.50              1  sample                41.000          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.90              1  sample                42.000          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.95              1  sample                42.000          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.99              1  sample                42.000          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.999             1  sample               166.000          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.9999            1  sample             11241.414          ns/op
ParseLongBenchMark.parseLong:parseLong·p1.00              1  sample            832512.000          ns/op
ParseLongBenchMark.parseLong                        1000000  sample  1204204       68.351 ± 8.961  ns/op
ParseLongBenchMark.parseLong:parseLong·p0.00        1000000  sample                   ≈ 0          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.50        1000000  sample                42.000          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.90        1000000  sample                83.000          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.95        1000000  sample                84.000          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.99        1000000  sample               125.000          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.999       1000000  sample               375.000          ns/op
ParseLongBenchMark.parseLong:parseLong·p0.9999      1000000  sample             24234.176          ns/op
ParseLongBenchMark.parseLong:parseLong·p1.00        1000000  sample           1861632.000          ns/op
```

conclusion
- the change of `Long.parseLong(obj.toString())` may have performance impact

### CODE REVIEW

File: tidb/src/main/java/io/tidb/bigdata/tidb/codec/RowEncoderV2.java [hunk 1]
```diff
@@ -199,7 +199,7 @@ private void encodeValue(CodecDataOutput cdo, Object value, DataType tp) {
       case TypeShort:
       case TypeTiny:
         // TODO: encode consider unsigned
-        encodeInt(cdo, (long) value);
+        encodeInt(cdo, Long.parseLong(value.toString()));
```
Comment (xuanyu66) [inline tidb/src/main/java/io/tidb/bigdata/tidb/codec/RowEncoderV2.java:202]: Can we remove .toString()

File: flink/flink-1.14/src/test/java/io/tidb/bigdata/flink/tidb/sink/TiKVSinkTypeTest.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+/**
+ * column type - MYSQLTYPE: int - TypeLong, bit-TypeBit, tinyint-TypeTiny, smallint-TypeShort
+ * mediumint-TypeInt24, year-TypeYear
+ */
+@Category(IntegrationTest.class)
```
Comment (xuanyu66) [inline flink/flink-1.14/src/test/java/io/tidb/bigdata/flink/tidb/sink/TiKVSinkTypeTest.java:47]: Change to Streaming group

**Review by xuanyu66 — CHANGES_REQUESTED** [review]

**Comment by xuanyu66** [comment]
Since it's a performance-sensitive change. 
https://jenkov.com/tutorials/java-performance/jmh.html

**Review by xuanyu66 — APPROVED** [review]

**Review by humengyu2012 — APPROVED** [review]

**Pull request merged**